gcs/blockcf2/blockcf.go

```diff
@@ -334,7 +334,13 @@ func Regular(block *wire.MsgBlock, prevScripts PrevScripter) (*gcs.FilterV2, err
 		// on consensus to make sure treasury opcodes never make it
 		// here. Setting it to true also will automatically enable the
 		// opcodes once the treasury agenda is voted in.
-		switch stake.DetermineTxType(tx, true) {
+		//
+		// Pass false for whether the automatic ticket revocations agenda is active.
+		// This is okay since the auto revocations flag just adds additional rules
+		// that revocations must have an empty signature script for its input and
+		// must have zero fee.  Revocations that don't follow those rules will still
+		// be rejected by consensus.
+		switch stake.DetermineTxType(tx, true, false) {
```

Comment:
This looks incorrect to me.  With `false` here, it will no longer recognize the new style of revocations once they activate and become the norm, which means none of their outputs scripts will end up in the filter.  That in turn will cause lightweight clients (like dcrwallet) that rely on the filters in order to detect the refunds to no longer be able to detect them and thus result in incorrect balances.  More generally, it would make the revocations invisible to lightweight clients.

It can't just be `true` either, because then the reverse case would happen.  That is to say it wouldn't add the outputs of revocations in the current format.

It also can't just look at the version of the transaction and set the flag based on that because version 2 revocation transactions are currently not rejected, so that would allow an unvoted consensus fork versus nodes running the old software.

Unfortunately, I really don't see a way around passing the flag into the function to pass along here.

The reason this trick worked with the treasury transactions is because they were invalid prior to the vote so consensus would reject them.  That doesn't apply here as described.

Reply:
I did just think of a possible alternative.  This could implement the revocation transaction detection directly here in such a way that it recognizes both formats regardless.  Then it could rely on the fact that consensus will be rejecting otherwise invalid revocations.

Reply:
> This looks incorrect to me. With false here, it will no longer recognize the new style of revocations once they activate and become the norm, which means none of their outputs scripts will end up in the filter.

It will still detect the new style of revocations.  The result of passing `false` here is just that [these checks](https://github.com/decred/dcrd/blob/5bc612042ad8eededc9a160b120c3c5e74fdc7e7/blockchain/stake/staketx.go#L1225-L1242) will not be run whether the agenda is active or not:
```
// Additional checks if the automatic ticket revocations agenda is active.
if isAutoRevocationsEnabled {
  // The input must have an empty signature script.
  if len(tx.TxIn[0].SignatureScript) != 0 {
    return stakeRuleError(ErrSSRtxInputHasSigScript, "SSRtx input 0 "+
      "contains a non-empty signature script")
  }


  // The fee must be zero.
  outputAmt := int64(0)
  for _, txOut := range tx.TxOut {
    outputAmt += txOut.Value
  }
  inputAmt := tx.TxIn[0].ValueIn
  if outputAmt < inputAmt {
    return stakeRuleError(ErrSSRtxInvalidFee, "SSRtx has a non-zero fee")
  }
}
```

The impact of those checks not running is that revocations with an empty signature script in its input or a non-zero fee will still (incorrectly) be identified as revocation transactions.  But I don't see any case where a revocation transaction would not be identified as a revocation here.

Reply:
After looking at the updated revocation function, I agree with your assessment.

I was fairly certain I remembered `CheckSSRtx` requiring the scripts to be reasonably close to being accurate, and looking at the code I see that I was mostly remembering correctly, but it's only [for the output scripts](https://github.com/decred/dcrd/blob/5bc612042ad8eededc9a160b120c3c5e74fdc7e7/blockchain/stake/staketx.go#L1217-L1222) versus the signature script.  

That check would be an issue if it applied to the signature script as I was misremembering when I originally commented,  however, since the existing checks only apply to the output scripts and the new checks are only adding more restrictive conditions to fields that are otherwise uninhibited until later by the consensus code, both old and new style revocations will be detected.